api/internal/managers/app/release/template.go

```diff
@@ -81,7 +81,13 @@ func (m *appReleaseManager) templateHelmChartCRs(configValues types.AppConfigVal
 		}
 
 		// Execute the template with config values
-		templatedYAML, err := m.templateEngine.Execute(configValues, template.WithProxySpec(proxySpec))
+		execOptions := []template.ExecOption{
+			template.WithProxySpec(proxySpec),
+		}
+		if registrySettings != nil {
```

Comment:
nit: why does this need an if statement? i think we should be consistent with the proxySpec above. it would still work the same if it's nil